server/icat/src/icatHighLevelRoutines.cpp

```diff
@@ -89,12 +90,17 @@ int chlOpen() {
     // cache the database type for subsequent calls
     try {
         const auto& database_plugin_map = irods::get_server_property<const nlohmann::json&>(std::vector<std::string>{irods::KW_CFG_PLUGIN_CONFIGURATION, irods::KW_CFG_PLUGIN_TYPE_DATABASE});
-        if ( database_plugin_map.size() != 1 ) {
-            rodsLog( LOG_ERROR, "Database plugin map must contain exactly one plugin object" );
+        if (database_plugin_map.empty()) {
+            rodsLog(LOG_ERROR, "%s: [database] configuration stanza needs attention.", __func__);
             return SYS_INVALID_INPUT_PARAM;
         }
-        database_plugin_type = database_plugin_map.items().begin().key();
-    } catch ( const irods::exception& e ) {
+        database_plugin_type = database_plugin_map.at(irods::KW_CFG_DB_TECHNOLOGY).get_ref<const std::string&>();
+    }
+    catch (const nlohmann::json::exception& e) {
+        rodsLog(LOG_ERROR, "%s: nlohmann json exception [%s]", __func__, e.what());
+        return SYS_INVALID_INPUT_PARAM;
+    }
```

Comment:
Please add a catch block for `const std::exception&`. It can return `SYS_LIBRARY_ERROR`.

Run a single unit test or core test to make sure everything is still good.

Reply:
A catch in the catch block, or a catch below (i.e. irods::exception -> json::exception -> std::exception)?

Reply:
Just an additional catch-block at the end.

irods::exception -> json::exception -> std::exception

Reply:
Added, message could probably be better though.